### PR TITLE
8281520: JFR: A wrong parameter is passed to the constructor of LeakKlassWriter

### PR DESCRIPTION
**Pull request by D-D-H**
Hi team,

Could I have a review of this fix?

The following code(jfrTypeSet.cpp:L388) passes a wrong parameter `_artifacts` to the constructor of LeakKlassWriter:
```
 387 } else {
 388 LeakKlassWriter lkw(_leakp_writer, _artifacts, _class_unload);
 389 CompositeKlassWriter ckw(&lkw, &kw);
 390 CompositeKlassWriterRegistration ckwr(&ckw, &reg);
 391 CompositeKlassCallback callback(&ckwr);
 392 _subsystem_callback = &callback;
 393 do_klasses();
 394 }
```

This problem is introduced by JDK-8225797 and fixed by JDK-8233111.

I filed a new issue to fix it since I think backporting JDK-8233111 is not necessary for the moment.

testing: test/jdk/jdk/jfr

Thanks,
Denghui

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8281520](https://bugs.openjdk.java.net/browse/JDK-8281520): JFR: A wrong parameter is passed to the constructor of LeakKlassWriter

### Reviewers
 * [Paul Hohensee](https://openjdk.java.net/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/813/head:pull/813` \
`$ git checkout pull/813`

Update a local copy of the PR: \
`$ git checkout pull/813` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/813/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 813`

View PR using the GUI difftool: \
`$ git pr show -t 813`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/813.diff">https://git.openjdk.java.net/jdk11u-dev/pull/813.diff</a>

</details>

### CODE REVIEW

**Review by phohensee — APPROVED** [review]
Lgtm.

**Comment by D-D-H** [comment]
@phohensee 
Thank you for the review.

Is it need a second review? If not, could you help sponsor it?

**Comment by phohensee** [comment]
No need for another review, but you have to go through the usual process of tagging the issue, adding a Fix Request comment, and getting it approved before I can sponsor it.

**Comment by D-D-H** [comment]
> 

Okay, thanks.

**Comment by GoeLin** [comment]
@D-D-H 
Please follow the instructions for backports: https://wiki.openjdk.java.net/display/JDKUpdates/How+to+contribute+a+fix
See point 6. You should only add the tag _after_ writing the corresponding comment in the JBS issue.

**Comment by D-D-H** [comment]
Hi @GoeLin ,

Thanks for the reminder.
But this fix is a new issue rather than a backport, do I still need to add a comment?

**Comment by GoeLin** [comment]
@D-D-H 
yes, please.
Any change that goes to jdk11u-dev needs a jdk11u-fix-yes by the maintainers.
The comment is the information we need to make a profound decision. It is your
way to address the maintainers.  The PR text addresses the reviewers (if you need any).
Maybe I need to improve the documentation, I'll have a look.

**Comment by D-D-H** [comment]
@GoeLin ,
Thanks, "Fix Request" is added.

**Comment by D-D-H** [comment]
> No need for another review, but you have to go through the usual process of tagging the issue, adding a Fix Request comment, and getting it approved before I can sponsor it.

Got approved, please help sponsor it when you have time, thanks in advance!

**Comment by phohensee** [comment]
Please add a "/integrate" comment. Then I can sponsor it.

**Pull request closed without merging**
